### PR TITLE
React components in TypeScript

### PR DESCRIPTION
**Pull request by ellthompson**
This PR rewrites the PCUI react components using typescript. The type definitions for our react builds can now be generated from the TypeScript compiler.

Each react component extends the `ElementComponent` class, which has an extendable and generic type for its properties of `ElementComponentProps`. Each component classes' props interface then extends the `ElementComponentProps` interface in order to receive a base set of props common to all components and then adds their own additional props. Each component props interface can also extend other common props interfaces (`IFlexProps`, `IParentProps`, `IResizableProps`) based on the components underlying functionality.

This PR also outputs module builds of PCUI using a preserved module directory structure to support tree shaking.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -94,5 +97,9 @@
   },
   "directories": {
     "doc": "docs"
+  },
+  "dependencies": {
+    "rollup-plugin-typescript2": "^0.34.1",
+    "typescript": "^4.8.4"
```
Comment (willeastcott) [inline package.json:103]: Why aren't these `devDependencies`?

File: src/CurveInput/index.js [hunk 2]
```diff
@@ -501,33 +500,33 @@ class CurveInput extends Container {
...
-        });
-        resetZoomTooltip.attach({
-            target: this._resetZoomButton
-        });
+        // var resetZoomTooltip = new Tooltip({
+        //     title: 'Reset Zoom'
+        // });
+        // resetZoomTooltip.attach({
+        //     target: this._resetZoomButton
+        // });
```
Comment (willeastcott) [inline src/CurveInput/index.js:508]: Any reason not to delete all this commented out code?

**Review by willeastcott — APPROVED** [review]
This is really great - nice job!

File: src/CurveInput/index.js (hunk 2 above)
Comment (ellthompson) [reply src/CurveInput/index.js:508]: As this is in an unused component, perhaps we should remove the entire file for now until it's ready to be worked on?

File: package.json (hunk 1 above)
Comment (ellthompson) [reply package.json:103]: I'll move them over, they were installed with --save.

**Pull request merged**